### DISCUSSION TITLE
[WinUI] Cannot customize FlyoutView easily

### DISCUSSION THREAD

**Discussion by mhrastegari** [dotnet/maui#21235] [Q&A]
## Summary

I wanted to use MAUI's NavigationView but I've got a few challenges:

- Cannot (or I don't know how to) change the SelectedItem color from blue(AccentColor)

![image](https://github.com/dotnet/maui/assets/42671084/c8797cac-337f-4434-8cc9-04fb64eff844)

- Items have a bit extra padding (not match with WinUI's style) and when I set `NavigationViewPaneDisplayMode.Auto` they do not have a good looking!

![image](https://github.com/dotnet/maui/assets/42671084/9e35237e-f966-4bbd-b6f4-91609c9ef575)

- The `PaneToggleButton` size is not match with WinUI's style and also when I resize the window it tries to be its default size 

![image](https://github.com/dotnet/maui/assets/42671084/5a195cb8-37f8-415b-b99b-7139eff9e28e)

- I'm not sure it should be handled in MAUI or WinAppSDK but when I open and close the menu I can see the background behind the ContentPage on the right side of the window!

https://github.com/dotnet/maui/assets/42671084/df349e66-a0c1-412c-b114-fa7a7bead583

- When I use `PaneToggleButton` to open and close the FlyoutView, FlyoutMenuItems won't get resized

https://github.com/dotnet/maui/assets/42671084/c2a34037-0811-4164-8801-d8f753610b2c

 ## Code to Repro

### Xaml
```xaml
<Shell
    x:Class="AdvancedNavigationView.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:AdvancedNavigationView"
    x:Name="flyout"
    FlyoutBackgroundColor="Transparent"
    FlyoutBehavior="{OnPlatform WinUI=Locked, Default=Flyout}">
    <ShellContent
        Title="Home"
        ContentTemplate="{DataTemplate local:MainPage}"
        Route="MainPage">
        <ShellContent.Icon>
            <FontImageSource FontFamily="Segoe Fluent Icons" Glyph="&#xe80f;" />
        </ShellContent.Icon>
    </ShellContent>
</Shell>
```
### C#
```csharp
    public partial class AppShell : Shell
    {
        public AppShell()
        {
            InitializeComponent();

            SetupNavigationView();
        }

        private void SetupNavigationView()
        {
#if WINDOWS
            Loaded += async delegate
            {
                var navigationView = (Microsoft.UI.Xaml.Controls.NavigationView)flyout.Handler!.PlatformView!;
                navigationView.IsPaneToggleButtonVisible = true;
                navigationView.PaneDisplayMode = Microsoft.UI.Xaml.Controls.NavigationViewPaneDisplayMode.Auto;
            };
#endif
        }
    }
```

## Suggestion
I think it's better to styling Native controls like Flyout with public properties (Like you guys did beautifully in App.xaml 💖) instead of doing it internally in MAUI :)
This saves a lot of time from both sides!

**Comment by PureWeen** [comment]
To style at that level I would recommend modifying the resource keys in your WinUI App.Xaml file

/Platform/Windows/App.Xaml

For example, to style that indicator you'd add this key

https://stackoverflow.com/a/49089606

**Reply by mhrastegari** [reply]
Thanks! is there a way to reset the size and paddings of FlyoutItem too? I didn't find any 💔
